Update about page (#255)
servo/servo.org: about.md

<h1>{{ title }}</h1>
  <p class="subtitle">
    Servo aims to empower developers with a <strong>lightweight, high-performance alternative for embedding web technologies</strong> in applications.
  </p>
<br>

> Created by Mozilla Research in 2012, the Servo project was started as a research and development effort. Stewardship of Servo moved from Mozilla Research to the [**Linux Foundation**](https://www.linuxfoundation.org/) in 2020, where its mission remains unchanged. In 2023 the project moved to [**Linux Foundation Europe**](https://linuxfoundation.eu/) trying to regain interest from a broader set of the industry.

Reviewer: ```suggestion
Created by Mozilla Research in 2012, the Servo project began as a research and development effort. Stewardship of Servo moved from Mozilla Research to the [**Linux Foundation**](https://www.linuxfoundation.org/) in 2020. The project finally moved to [**Linux Foundation Europe**](https://linuxfoundation.eu/) in 2023 with the mission of trying to regain interest from the broader industry.
```

Mainly I just wanted to remove the 'its mission remains unchanged' line.

Servo is written in [**Rust**](https://www.rust-lang.org/), taking advantage of the memory safety properties and concurrency features of the language.

Servo is an **independent** project hosted at Linux Foundation Europe and managed openly by the [Technical Steering Committee](https://github.com/servo/project/blob/main/governance/README.md).

> We are currently working on **transitioning Servo from a research project to a production-ready web engine** by delivering stable APIs, improving user interaction capabilities, and fostering an active community that contributes to its growth.

Reviewer: ```suggestion
The current short-term focus of the project is to **transition Servo from a research project to a production-ready web engine** by delivering stable APIs, improving user interaction capabilities, and fostering an active community that contributes to its growth.
```

It might be better to rewrite this paragraph and the next one in the passive voice to match the rest of the document.

> In the long run, we want to make Servo the **leading embeddable web engine** that powers innovative applications and frameworks, offering unparalleled performance, stability, and modern web standards compliance.

Reviewer: ```suggestion
The project's long-term vision is to make Servo the **leading embeddable web engine** that powers innovative applications and frameworks, offering unparalleled performance, stability, and modern web standards compliance.
```
